firebase.json

```diff
@@ -24,1610 +24,648 @@
+        "destination": "https://amp.dev/:lang/documentation/guides-and-tutorials/integrate/integrate-with-apps",
         "type": 301
       },
       {
-        "source": "/:lang/docs/guides/responsive/responsive_design",
-        "destination": "/:lang/docs/design/responsive/responsive_design.html",
+        "source": "/:lang/docs/integration/integrate-your-tech",
+        "destination": "https://amp.dev/:lang/documentation/guides-and-tutorials/contribute/integrate-your-tech",
         "type": 301
       },
       {
-        "source": "/:lang/docs/guides/responsive/style_pages",
-        "destination": "/:lang/docs/design/responsive/style_pages.html",
+        "source": "/:lang/docs/integration/pwa-amp",
+        "destination": "https://amp.dev/:lang/documentation/guides-and-tutorials/integrate/amp-in-pwa",
         "type": 301
       },
       {
-        "source": "/:lang/docs/guides/third_party_components",
-        "destination": "/:lang/docs/fundamentals/third_party_components.html",
+        "source": "/:lang/docs/integration/pwa-amp/amp-as-pwa",
+        "destination": "https://amp.dev/:lang/documentation/guides-and-tutorials/optimize-and-measure/amp-as-pwa",
         "type": 301
       },
       {
-        "source": "/:lang/docs/guides/validate",
-        "destination": "/:lang/docs/fundamentals/validate.html",
+        "source": "/:lang/docs/integration/pwa-amp/amp-in-pwa",
+        "destination": "https://amp.dev/:lang/documentation/guides-and-tutorials/integrate/amp-in-pwa",
         "type": 301
       },
       {
-        "source": "/:lang/docs/reference/spec",
-        "destination": "/:lang/docs/fundamentals/spec.html",
+        "source": "/:lang/docs/integration/pwa-amp/amp-to-pwa",
+        "destination": "https://amp.dev/:lang/documentation/guides-and-tutorials/integrate/amp-to-pwa",
         "type": 301
       },
       {
-        "source": "/:lang/docs/reference/spec/amp-boilerplate",
-        "destination": "/:lang/docs/fundamentals/spec/amp-boilerplate.html",
+        "source": "/:lang/docs/integration/supported_integrations",
+        "destination": "https://amp.dev/:lang/community/platform-and-vendor-partners.html",
         "type": 301
       },
       {
-        "source": "/:lang/docs/reference/spec/amp-html-layout",
-        "destination": "/:lang/docs/design/amp-html-layout.html",
+        "source": "/:lang/docs/interaction_dynamic/",
+        "destination": "https://amp.dev/:lang/documentation/guides-and-tutorials/",
         "type": 301
       },
       {
-        "source": "/:lang/docs/reference/validation_errors",
-        "destination": "/:lang/docs/troubleshooting/validation_errors.html",
+        "source": "/:lang/docs/interaction_dynamic/interactivity",
+        "destination": "https://amp.dev/:lang/documentation/guides-and-tutorials/develop/interactivity/create-interactive.html",
         "type": 301
       },
       {
-        "source": "/:lang/docs/tutorials/add
```

Comment:
Rereading the documentation this probably needs to be `**`, doesn't it? See: https://firebase.google.com/docs/hosting/full-config#redirects